forge-game/src/main/java/forge/game/ability/effects/DestroyAllEffect.java

```diff
@@ -89,10 +89,11 @@ public void resolve(SpellAbility sa) {
 
         list = GameActionUtil.orderCardsByTheirOwners(game, list, ZoneType.Graveyard, sa);
 
-        CardZoneTable table = new CardZoneTable();
         Map<AbilityKey, Object> params = AbilityKey.newMap();
         params.put(AbilityKey.LastStateBattlefield, game.copyLastStateBattlefield());
 
+        CardZoneTable table = new CardZoneTable(game.getLastStateBattlefield(), game.getLastStateGraveyard());
```

Comment:
I need to update all other places
but in tests it already works with something like _Ruination_